cetlvast/suites/unittest/test_variable_length_array_bool.cpp

```diff
@@ -231,4 +231,112 @@ TYPED_TEST(VLABoolTests, TestBoolIterator)
     ASSERT_EQ(false, *foo.cbegin());
     ASSERT_EQ(true, *(foo.cend() - 1));
     ASSERT_EQ(true, foo.cbegin()[2]);
+
+    typename decltype(foo)::const_iterator const_begin_iterator =
+        const_cast<typename std::add_pointer<typename std::add_const<decltype(foo)>::type>::type>(&foo)->begin();
+    typename decltype(foo)::const_iterator const_end_interator =
+        const_cast<typename std::add_pointer<typename std::add_const<decltype(foo)>::type>::type>(&foo)->end();
+
+    ASSERT_EQ(false, *const_begin_iterator);
+    ASSERT_EQ(true, *(const_end_interator - 1));
+    ASSERT_EQ(true, const_begin_iterator[2]);
```

Comment:
Is this correct?  foo at index 2 is false, not true.

Reply:
This is a bool array. There is no 2 unless you are running on an unbalanced ternary machine. Unfortunately CETL does not support trit boolean values. Sorry.

Reply:
Oh I missed that on line 193 the iterator is on index 2 and it sets the value to true.  Nevermind.